applications/tari_conduit/src/main.rs

```diff
@@ -0,0 +1,630 @@
+                    "{",
+                    "{",
+                    e.code(),
+                    e.description(),
+                    "}",
+                    "}"
+                )
+                .as_bytes()
+                .to_vec();
+            },
+        }
+    }
+    data
+}
+
+// Structure http header and body
+// TODO: Error headers
+fn structure_response(response_data: &[u8], code: u32) -> String {
+    let header = format!(
+        "HTTP/1.1 {} \
+         OK\r\nAccept-Ranges:bytes\r\nContent-Length:{}\r\nContent-Type:application/json\r\nServer:Epee-based\r\n\r\n",
+        code,
+        String::from_utf8_lossy(response_data).len()
+    );
+    format!("{}{}", header, String::from_utf8_lossy(response_data))
+}
+
+// Get the url bits from http header
+fn get_url_part(request: &[u8]) -> String {
+    let string = String::from_utf8_lossy(&request[..]).to_string();
+    let mut split_request = string.lines();
+    let first_line = split_request.next().unwrap().to_string();
+    let mut iter = first_line.split_whitespace();
+    iter.next();
+    iter.next().unwrap().to_string()
+}
+
+// Get the request type from http header
+fn get_request_type(request: &[u8]) -> String {
+    let string = String::from_utf8_lossy(&request[..]).to_string();
+    let mut split_request = string.lines();
+    let first_line = split_request.next().unwrap().to_string();
+    let mut iter = first_line.split_whitespace();
+    iter.next().unwrap().to_string()
+}
+
+// Extract json from response
+fn get_json(request: &[u8]) -> Option<Vec<u8>> {
+    let re = Regex::new(r"\{(.*)\}").unwrap(); // Match text from first '{' to last '}'
+    let string = stringify_request(request);
+    let caps = re.captures(&string);
+    match caps {
+        Some(caps) => {
+            match caps.get(0) {
+                Some(json) => {
+                    let result = json.as_str().as_bytes().to_vec();
+                    Some(result)
+                },
+                None => {
+                    // Request was malformed.
+                    debug!("Malformed Request");
+                    None
+                },
+            }
+        },
+        None => {
+            // Request didn't contain any json.
+            debug!("No Request");
+            debug!("Request: {}", string);
+            None
+        },
+    }
+}
+
+// Convert json bytes to json string
+fn stringify_request(buffer: &[u8]) -> String {
+    String::from_utf8_lossy(&buffer).to_string()
+}
+
+// Get method name from post request
+fn get_post_json_method(json: &[u8]) -> String {
+    let parsed = json::parse(&stringify_request(json)).unwrap();
+    debug!("{}", parsed);
+    if !parsed["method"].is_null() {
+        return parsed["method"].to_string();
+    }
+    "".to_string()
+}
+
+// Utility function to transform array to fixed array
+fn from_slice(bytes: &[u8]) -> [u8; 32] {
```

Comment:
I think some of these could be used as a utiliy class , and kept seperate

Reply:
Agreed, could probably put them into tari_utilities perhaps?